web-services/security/src/main/java/datawave/security/login/DatawavePrincipalLoginModule.java

```diff
@@ -412,94 +426,114 @@ protected DatawaveCredential getDatawaveCredential() throws LoginException {
             identity = unauthenticatedIdentity;
             log.trace("Authenticating as unauthenticatedIdentity=" + identity);
         }
-        if (trace)
+        if (trace) {
             log.trace("identity = " + identity);
+        }
         if (identity == null) {
-
             if (credential.getCertificate() != null || (!trustedHeaderLogin && !jwtHeaderLogin)) {
                 if (!validateCertificateCredential(credential)) {
                     log.debug("Bad credential for alias=" + credential.getUserName());
-                    throw new FailedLoginException("Supplied Credential did not match existing credential for " + credential.getUserName());
+                    throw new CredentialException("Validation of credential failed");
                 }
             }
 
             if (!jwtHeaderLogin || credential.getJwtToken() == null) {
                 try {
                     identity = new DatawavePrincipal(datawaveUserService.lookup(credential.getEntities()));
+                } catch (AuthorizationException e) {
+                    Throwable cause = e.getCause();
+                    String message = cause != null ? cause.getMessage() : e.getMessage();
+                    log.debug("Failing login due to datawave user service exception " + e.getMessage(), e);
+                    // should result in a SERVICE_UNAVAILABLE (503) response code in DatawaveAuthenticationMechanism.sendChallenge
+                    throw new LoginException("Unable to authenticate: " + message);
                 } catch (Exception e) {
                     log.debug("Failing login due to datawave user service exception " + e.getMessage(), e);
-                    throw new FailedLoginException("Unable to authenticate: " + e.getMessage());
+                    // should result in a SERVICE_UNAVAILABLE (503) response code in DatawaveAuthenticationMechanism.sendChallenge
+                    throw new LoginException("Unable to authenticate: " + e.getMessage());
                 }
             } else {
                 try {
                     identity = new DatawavePrincipal(jwtTokenHandler.createUsersFromToken(credential.getJwtToken()));
                 } catch (Exception e) {
                     log.debug("Failing login due to JWT token exception " + e.getMessage(), e);
-                    throw new FailedLoginException("Unable to authenticate: " + e.getMessage());
+                    // should result in an UNAUTHORIZED (501) response code in DatawaveAuthenticationMechanism.sendChallenge
+                    throw new CredentialException("Unable to authenticate: " + e.getMessage());
                 }
             }
         }
         if (getUseFirstPass()) {
             sharedState.put("javax.security.auth.login.name", alias);
             sharedState.put("javax.security.auth.login.password", credential.getCertificate());
         }
-        if (trace)
+        if (log.isTraceEnabled()) {
```

Comment:
I could be missing something, but unclear why the switch from `if (trace)` to `if (log.isTraceEnabled())`